src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Lottery {
+    private Ball ball;
+    private ColorSupplier color;
+    private NumberBall numberBall;
+
+    public Lottery(Ball ball, ColorSupplier color, NumberBall numberBall) {
+        this.ball = ball;
+        this.color = color;
+        this.numberBall = numberBall;
```

Comment:
The constructor should not be used to set a specific ball instance. Instead, create a new ball instance within the 'getRandomBall()' method.

Reply:
moved creation this.ball = new Ball()
to method Ball getRandomBall()